.github/workflows/spark_test.yaml

```diff
@@ -46,17 +46,23 @@ jobs:
           export PATH="~/.pyenv/bin:$PATH"
           eval "$(pyenv init -)"
           eval "$(pyenv virtualenv-init -)"
-          pyenv install 3.7.4
-          pyenv global system 3.7.4
-          pipenv --python 3.7 install
-          pipenv run pip install pyspark==3.4.0
+          pyenv install 3.8.18
+          pyenv global system 3.8.18
+          pipenv --python 3.8 install
+          # pipenv run pip install pyspark==3.4.0
+          # Download and install pyspark from staged 3.5.0 RC candidate
+          pipenv run pip install https://dist.apache.org/repos/dist/dev/spark/v3.5.0-rc4-bin/pyspark-3.5.0.tar.gz
           pipenv run pip install flake8==3.5.0 pypandoc==1.3.3
           pipenv run pip install importlib_metadata==3.10.0
-          pipenv run pip install mypy==0.910
+          pipenv run pip install mypy==0.982
           pipenv run pip install cryptography==37.0.4
           pipenv run pip install twine==4.0.1
           pipenv run pip install wheel==0.33.4
           pipenv run pip install setuptools==41.0.1
+          pipenv run pip install pydocstyle==3.0.0
+          pipenv run pip install pandas==1.0.5
```

Comment:
`pandas`, `pyarrow` and `numpy` dependencies are needed because of this change: https://github.com/apache/spark/commit/7c1ad5bb60c88c1c659c131e5119aab1f8212af5#diff-2730c6c75b3ee4fc611a978044e949e2f5a92238ab1ac78ab1bd268d51930251L19